### PR TITLE
feat(eventarc): add sample for unmarshalling Cloud Storage event

### PR DESCRIPTION
**Pull request by grayside**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: eventarc/storage_handler/handler.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/googleapis/google-cloudevents-go/cloud/storagedata"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// HelloStorage receives and processes a CloudEvent containing StorageObjectData
+func HelloStorage(w http.ResponseWriter, r *http.Request) {
+	ce, err := cloudevents.NewEventFromHTTPRequest(r)
+	if err != nil {
+		log.Printf("failed to parse CloudEvent: %v", err)
+		msg := fmt.Sprintf("%s: expected CloudEvent", http.StatusText(http.StatusBadRequest))
```
Comment (muncus) [inline eventarc/storage_handler/handler.go:35]: (optional): i don't think the `http.StatusText()` is adding much here.
If we use a string literal (even with the same "Bad Request" text), it saves a line in each of these error clauses, making a more concise sample.

File: eventarc/storage_handler/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	"testing"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/googleapis/google-cloudevents-go/cloud/storagedata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// [START eventarc_testing_cloudevent]
```
Comment (muncus) [inline eventarc/storage_handler/handler_test.go:34]: can we move this region tag up so the imports are included?

**Review by muncus — APPROVED** [review]

File: eventarc/storage_handler/handler.go (hunk 1 above)
Comment (grayside) [reply eventarc/storage_handler/handler.go]: Done.

File: eventarc/storage_handler/handler_test.go (hunk 2 above)
Comment (grayside) [reply eventarc/storage_handler/handler_test.go]: Done.

**Comment by grayside** [comment]
In addition to addressing feedback, I added:
* a success test case for an HTTP 200 response code
* removed scoping tricks to reuse a `want` variable, instead switched those to `wantStatus` & `wantBody` variables.

**Pull request merged**